Review: Update German translation for Git v2.36.0

ruester opened the pull request:
@ralfth and @PhillipSz: Please have a look at my suggestions for the German translation of Git v2.36.0. Only the commit e814280d10 needs to be reviewed, the commit aff6b0921f is the automatically generated one to update the po file.
As always I also updated some other translations that are not part of this Git update. For example omitting the full stop at the end of a sentence if the original string does not contain one or starting with a small letter if the original string does too...

carenas committed:
mingw: avoid fallback for {local,gm}time_r()

mingw-w64's pthread_unistd.h had a bug that mistakenly (because there is
no support for the *lockfile() functions required[1]) defined
_POSIX_THREAD_SAFE_FUNCTIONS and that was being worked around since
3ecd153a3b (compat/mingw: support MSys2-based MinGW build, 2016-01-14).

The bug was fixed in winphtreads, but as a side effect, leaves the
reentrant functions from time.h no longer visible and therefore breaks
the build.

Since the intention all along was to avoid using the fallback functions,
formalize the use of POSIX by setting the corresponding feature flag and
compile out the implementation for the fallback functions.

[1] https://unix.org/whitepapers/reentrant.html

Signed-off-by: Carlo Marcelo Arenas Belón <carenas@gmail.com>
Acked-by: Johannes Schindelin <Johannes.Schindelin@gmx.de>
Signed-off-by: Junio C Hamano <gitster@pobox.com>

dscho committed:
Add a function to determine whether a path is owned by the current user

This function will be used in the next commit to prevent
`setup_git_directory()` from discovering a repository in a directory
that is owned by someone other than the current user.

Note: We cannot simply use `st.st_uid` on Windows just like we do on
Linux and other Unix-like platforms: according to
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/stat-functions
this field is always zero on Windows (because Windows' idea of a user ID
does not fit into a single numerical value). Therefore, we have to do
something a little involved to replicate the same functionality there.

Also note: On Windows, a user's home directory is not actually owned by
said user, but by the administrator. For all practical purposes, it is
under the user's control, though, therefore we pretend that it is owned
by the user.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

dscho committed:
setup_git_directory(): add an owner check for the top-level directory

It poses a security risk to search for a git directory outside of the
directories owned by the current user.

For example, it is common e.g. in computer pools of educational
institutes to have a "scratch" space: a mounted disk with plenty of
space that is regularly swiped where any authenticated user can create
a directory to do their work. Merely navigating to such a space with a
Git-enabled `PS1` when there is a maliciously-crafted `/scratch/.git/`
can lead to a compromised account.

The same holds true in multi-user setups running Windows, as `C:\` is
writable to every authenticated user by default.

To plug this vulnerability, we stop Git from accepting top-level
directories owned by someone other than the current user. We avoid
looking at the ownership of each and every directories between the
current and the top-level one (if there are any between) to avoid
introducing a performance bottleneck.

This new default behavior is obviously incompatible with the concept of
shared repositories, where we expect the top-level directory to be owned
by only one of its legitimate users. To re-enable that use case, we add
support for adding exceptions from the new default behavior via the
config setting `safe.directory`.

The `safe.directory` config setting is only respected in the system and
global configs, not from repository configs or via the command-line, and
can have multiple values to allow for multiple shared repositories.

We are particularly careful to provide a helpful message to any user
trying to use a shared repository.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

dscho committed:
Git 2.30.3

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

dscho committed:
GIT-VERSION-GEN: bump to v2.33.1

This was missed in af6d1d602a8f (Git 2.33.1, 2021-10-12).

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

dscho committed:
Sync with 2.30.3

* maint-2.30:
  Git 2.30.3
  setup_git_directory(): add an owner check for the top-level directory
  Add a function to determine whether a path is owned by the current user

dscho committed:
Git 2.31.2

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

dscho committed:
Sync with 2.31.2

* maint-2.31:
  Git 2.31.2
  Git 2.30.3
  setup_git_directory(): add an owner check for the top-level directory
  Add a function to determine whether a path is owned by the current user

dscho committed:
Git 2.32.1

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

dscho committed:
Sync with 2.32.1

* maint-2.32:
  Git 2.32.1
  Git 2.31.2
  Git 2.30.3
  setup_git_directory(): add an owner check for the top-level directory
  Add a function to determine whether a path is owned by the current user

dscho committed:
Git 2.33.2

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

dscho committed:
Sync with 2.33.2

* maint-2.33:
  Git 2.33.2
  Git 2.32.1
  Git 2.31.2
  GIT-VERSION-GEN: bump to v2.33.1
  Git 2.30.3
  setup_git_directory(): add an owner check for the top-level directory
  Add a function to determine whether a path is owned by the current user

dscho committed:
Git 2.34.2

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

dscho committed:
Sync with 2.34.2

* maint-2.34:
  Git 2.34.2
  Git 2.33.2
  Git 2.32.1
  Git 2.31.2
  GIT-VERSION-GEN: bump to v2.33.1
  Git 2.30.3
  setup_git_directory(): add an owner check for the top-level directory
  Add a function to determine whether a path is owned by the current user

dscho committed:
Git 2.35.2

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

dscho committed:
Merge branch 'cb/mingw-gmtime-r'

Build fix on Windows.

* cb/mingw-gmtime-r:
  mingw: avoid fallback for {local,gm}time_r()

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

dscho committed:
Fix `GIT_CEILING_DIRECTORIES` with `C:\` and the likes

When determining the length of the longest ancestor of a given path with
respect to to e.g. `GIT_CEILING_DIRECTORIES`, we special-case the root
directory by returning 0 (i.e. we pretend that the path `/` does not end
in a slash by virtually stripping it).

That is the correct behavior because when normalizing paths, the root
directory is special: all other directory paths have their trailing
slash stripped, but not the root directory's path (because it would
become the empty string, which is not a legal path).

However, this special-casing of the root directory in
`longest_ancestor_length()` completely forgets about Windows-style root
directories, e.g. `C:\`. These _also_ get normalized with a trailing
slash (because `C:` would actually refer to the current directory on
that drive, not necessarily to its root directory).

In fc56c7b34b (mingw: accomodate t0060-path-utils for MSYS2,
2016-01-27), we almost got it right. We noticed that
`longest_ancestor_length()` expects a slash _after_ the matched prefix,
and if the prefix already ends in a slash, the normalized path won't
ever match and -1 is returned.

But then that commit went astray: The correct fix is not to adjust the
_tests_ to expect an incorrect -1 when that function is fed a prefix
that ends in a slash, but instead to treat such a prefix as if the
trailing slash had been removed.

Likewise, that function needs to handle the case where it is fed a path
that ends in a slash (not only a prefix that ends in a slash): if it
matches the prefix (plus trailing slash), we still need to verify that
the path does not end there, otherwise the prefix is not actually an
ancestor of the path but identical to it (and we need to return -1 in
that case).

With these two adjustments, we no longer need to play games in t0060
where we only add `$rootoff` if the passed prefix is different from the
MSYS2 pseudo root, instead we also add it for the MSYS2 pseudo root
itself. We do have to be careful to skip that logic entirely for Windows
paths, though, because they do are not subject to that MSYS2 pseudo root
treatment.

This patch fixes the scenario where a user has set
`GIT_CEILING_DIRECTORIES=C:\`, which would be ignored otherwise.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

steadmon committed:
ls-tree: `-l` should not imply recursive listing

In 9c4d58ff2c (ls-tree: split up "fast path" callbacks, 2022-03-23), a
refactoring of the various read_tree_at() callbacks caused us to
unconditionally recurse into directories if `-l` (long format) was
passed on the command line, regardless of whether or not we also pass
the `-r` (recursive) flag.

Fix this by making show_tree_long() return the value of `recurse`,
rather than always returning 1. This value is interpreted by
read_tree_at() to be a signal on whether or not to recurse.

Signed-off-by: Josh Steadmon <steadmon@google.com>
Signed-off-by: Junio C Hamano <gitster@pobox.com>

carenas committed:
git-compat-util: really support openssl as a source of entropy

05cd988dce5 (wrapper: add a helper to generate numbers from a CSPRNG,
2022-01-17), configure openssl as the source for entropy in NON-STOP
but doesn't add the needed header or link options.

Since the only system that is configured to use openssl as a source
of entropy is NON-STOP, add the header unconditionally, and -lcrypto
to the list of external libraries.

An additional change is required to make sure a NO_OPENSSL=1 build
will be able to work as well (tested on Linux with a modified value
of CSPRNG_METHOD = openssl), and the more complex logic that allows
for compatibility with APPLE_COMMON_CRYPTO or allowing for simpler
ways to link (without libssl) has been punted for now.

Reported-by: Randall Becker <rsbecker@nexbridge.com>
Signed-off-by: Carlo Marcelo Arenas Belón <carenas@gmail.com>
Signed-off-by: Junio C Hamano <gitster@pobox.com>

me-and committed:
configure.ac: fix HAVE_SYNC_FILE_RANGE definition

If sync_file_range is not available when building the configure script,
there is a cosmetic bug when running that script reporting
"HAVE_SYNC_FILE_RANGE: command not found".  Remove that error message by
defining HAVE_SYNC_FILE_RANGE to an empty string, rather than generating
a script where that appears as a bare command.

Signed-off-by: Adam Dinwoodie <adam@dinwoodie.org>
Signed-off-by: Junio C Hamano <gitster@pobox.com>

avar committed:
Documentation/Makefile: fix "make info" regression in dad9cd7d518

Fix a regression in my dad9cd7d518 (Makefile: move ".SUFFIXES" rule to
shared.mak, 2022-03-03). As explained in the GNU make documentation
for the $* variable, available at:

	info make --index-search='$*'

This rule relied on ".texi" being in the default list of suffixes, as
seen at:

	make -f/dev/null -p | grep -v -e ^# -e ^$|grep -F .SUFFIXES

The documentation explains what was going on here:

	In an explicit rule, there is no stem; so '$*' cannot be determined
	in that way.  Instead, if the target name ends with a recognized
	suffix (*note Old-Fashioned Suffix Rules: Suffix Rules.), '$*' is
	set to the target name minus the suffix.  For example, if the
	target name is 'foo.c', then '$*' is set to 'foo', since '.c' is a
	suffix.  GNU 'make' does this bizarre thing only for compatibility
	with other implementations of 'make'.  You should generally avoid
	using '$*' except in implicit rules or static pattern rules.

	If the target name in an explicit rule does not end with a
	recognized suffix, '$*' is set to the empty string for that rule.

I.e. this rule added back in 5cefc33bffd (Documentation: add
gitman.info target, 2007-12-10) was resolving gitman.texi from
gitman.info. We can instead just use the more obvious $< variable
referring to the prerequisite.

This was the only use of $* in our Makefiles in an explicit rule, the
three remaining ones are all implicit rules, and therefore didn't
depend on the ".SUFFIXES" list.

Reported-by: Adam Dinwoodie <adam@dinwoodie.org>
Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
Tested-by: Adam Dinwoodie <adam@dinwoodie.org>
Signed-off-by: Junio C Hamano <gitster@pobox.com>

vdye committed:
contrib/scalar: fix 'all' target in Makefile

Add extra ':' to second 'all' target definition to allow 'scalar' to build.
Without this fix, the 'all:' and 'all::' targets together cause a build
failure when 'scalar' build is enabled with 'INCLUDE_SCALAR':

    Makefile:14: *** target file `all' has both : and :: entries.  Stop.

Signed-off-by: Victoria Dye <vdye@github.com>
Signed-off-by: Junio C Hamano <gitster@pobox.com>

fangyi-zhou committed:
l10n: zh_CN v2.36.0 round 1

Reviewed-by: 依云 <lilydjwg@gmail.com>
Signed-off-by: Fangyi Zhou <me@fangyi.io>

fangyi-zhou committed:
l10n: Update zh_CN repo link

Signed-off-by: Fangyi Zhou <me@fangyi.io>

tmzullinger committed:
doc: replace "--" with {litdd} in credential-cache/fsmonitor

Asciidoc renders `--` as em-dash.  This is not appropriate for command
names.  It also breaks linkgit links to these commands.

Fix git-credential-cache--daemon and git-fsmonitor--daemon.  The latter
was added 3248486920 (fsmonitor: document builtin fsmonitor, 2022-03-25)
and included several links.  A check for broken links in the HTML docs
turned this up.

Manually inspecting the other Documentation/git-*--*.txt files turned up
the issue in git-credential-cache--daemon.

While here, quote `git credential-cache--daemon` in the synopsis to
match the vast majority of our other documentation.

Signed-off-by: Todd Zullinger <tmz@pobox.com>
Signed-off-by: Junio C Hamano <gitster@pobox.com>

fangyi-zhou committed:
submodule-helper: fix usage string

The missing space at the end of the line makes the closing square
bracket sticking to the dash in the next line

Found during localisation v2.36.0 round 1

Signed-off-by: Fangyi Zhou <me@fangyi.io>
Signed-off-by: Junio C Hamano <gitster@pobox.com>

gitster committed:
Merge branch 'ab/make-optim-noop'

A micro fix to a topic earlier merged to 'master'
source: <patch-1.1-05949221e3f-20220319T002715Z-avarab@gmail.com>

* ab/make-optim-noop:
  contrib/scalar: fix 'all' target in Makefile
  Documentation/Makefile: fix "make info" regression in dad9cd7d518

gitster committed:
Merge branch 'ns/core-fsyncmethod'

A couple of fix-up to a topic that is now in 'master'.
source: <pull.1193.git.1648663716891.gitgitgadget@gmail.com>

* ns/core-fsyncmethod:
  configure.ac: fix HAVE_SYNC_FILE_RANGE definition

gitster committed:
Merge branch 'bc/csprng-mktemps'

Build fix.

* bc/csprng-mktemps:
  git-compat-util: really support openssl as a source of entropy

gitster committed:
Merge branch 'tl/ls-tree-oid-only'

"git ls-tree" learns "--oid-only" option, similar to "--name-only",
and more generalized "--format" option.
source: <cover.1648026472.git.dyroneteng@gmail.com>

* tl/ls-tree-oid-only:
  ls-tree: `-l` should not imply recursive listing

Commit:
git-ls-tree.txt: fix the name of "%(objectsize:padded)"

Commit 455923e0a1 ("ls-tree: introduce "--format" option", 2022-03-23)
introduced `--format` and the various placeholders it can take, such as
%(objectname) and %(objectsize).

At some point when that patch was being developed, those placeholders
had shorter names, e.g., %(name) and %(size), which can be seen in the
commit message of 455923e0a1. One instance of "%(size:padded)" also
managed to enter the documentation in the final version of the patch.
Correct it to "%(objectsize:padded)".

Signed-off-by: Martin Ågren <martin.agren@gmail.com>
Signed-off-by: Junio C Hamano <gitster@pobox.com>

avar committed:
Documentation: add --batch-command to cat-file synopsis

440c705ea63 (cat-file: add --batch-command mode, 2022-02-18) added
the new option and operating mode without listing it to the synopsis
section.  Fix it.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
Signed-off-by: Junio C Hamano <gitster@pobox.com>

gitster committed:
Merge branch 'tl/ls-tree-oid-only'

* tl/ls-tree-oid-only:
  git-ls-tree.txt: fix the name of "%(objectsize:padded)"

gitster committed:
Merge branch 'js/apply-partial-clone-filters-recursively'

Typofix

* js/apply-partial-clone-filters-recursively:
  submodule-helper: fix usage string

gitster committed:
Merge branch 'tz/doc-litdd-fixes'

Documentation markup fix.

* tz/doc-litdd-fixes:
  doc: replace "--" with {litdd} in credential-cache/fsmonitor

gitster committed:
Merge branch 'jc/cat-file-batch-commands'

Doc fix.

* jc/cat-file-batch-commands:
  Documentation: add --batch-command to cat-file synopsis

me-and committed:
t9902: split test to run on appropriate systems

The "FUNNYNAMES" test prerequisite passes on Cygwin, as the Cygwin
file system interface has a workaround for the underlying operating
system's lack of support for tabs, newlines or quotes.  However, it does
not add support for backslash, which is treated as a directory
separator, meaning one of the tests added by 48803821b1 ("completion:
handle unusual characters for sparse-checkout", 2022-02-07) will fail on
Cygwin.

To avoid this failure while still getting maximal test coverage, split
that test into two: test handling of paths that include tabs on anything
that has the FUNNYNAMES prerequisite, but skip testing handling of paths
that include backslashes unless both FUNNYNAMES is set and the system is
not Cygwin.

It might be nice to have more granularity than "FUNNYNAMES" and its
sibling "FUNNIERNAMES" provide, so that tests could be run based on
specific individual characters supported by the file system being
tested, but that seems like it would make the prerequisite checks in
this area much more verbose for very little gain.

Signed-off-by: Adam Dinwoodie <adam@dinwoodie.org>
Signed-off-by: Junio C Hamano <gitster@pobox.com>

avar committed:
ls-tree doc: document interaction with submodules

The ls-tree documentation had never been updated after it learned to
interact with submodules to explicitly mention them. The initial
support was added in f35a6d3bce7 (Teach core object handling functions
about gitlinks, 2007-04-09). E.g. the discussion of --long added in
f35a6d3bce7 (Teach core object handling functions about gitlinks,
2007-04-09) didn't explicitly mention them.

But this documentation added in 455923e0a15 (ls-tree: introduce
"--format" option, 2022-03-23) had no such excuse, and was actively
misleading by providing an exhaustive but incomplete list of object
types we'd emit.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
Signed-off-by: Junio C Hamano <gitster@pobox.com>

gitster committed:
Merge branch 'tl/ls-tree-oid-only'

Docfix.

* tl/ls-tree-oid-only:
  ls-tree doc: document interaction with submodules

gitster committed:
Merge branch 'ld/sparse-index-bash-completion'

Test regression fix.

* ld/sparse-index-bash-completion:
  t9902: split test to run on appropriate systems

gitster committed:
Git 2.36-rc1

Signed-off-by: Junio C Hamano <gitster@pobox.com>

ruester committed:
Update de.po for new translations

Signed-off-by: Matthias Rüster <matthias.ruester@gmail.com>

jnavila committed:
l10n: fr: v2.36 round 1

Signed-off-by: Jean-Noël Avila <jn.avila@free.fr>

ruester committed:
l10n: de.po: Update German translation

Signed-off-by: Matthias Rüster <matthias.ruester@gmail.com>

ralfth commented on po/de.po:
```diff
@@ -555,11 +555,11 @@ msgstr "Konnte farbigen Block-Header '%.*s' nicht parsen."
 #: add-patch.c:431
 msgid "could not parse diff"
-msgstr "Konnte Differenz nicht parsen."
+msgstr "konnte Differenz nicht parsen"
```
Maybe we should translate it as "Diff" (der Diff).  We already do so in other messages.

ruester replied:
ok, I will also change it in our "glossary"

ralfth commented on po/de.po:
```diff
@@ -555,11 +555,11 @@ msgstr "Konnte farbigen Block-Header '%.*s' nicht parsen."
 #: add-patch.c:450
 msgid "could not parse colored diff"
-msgstr "Konnte farbige Differenz nicht parsen."
+msgstr "konnte farbige Differenz nicht parsen"
```
Same here..

ralfth commented on po/de.po:
```diff
@@ -6118,36 +6130,36 @@ msgstr "%s: %s - %s"
+#, c-format
 msgid "%s tree"
-msgstr "Thread %s fehlgeschlagen."
+msgstr "%s Verzeichnis"
```
I think this should be "%s Tree" as it's referring to a tree object.

ruester replied:
I've now added it also to https://github.com/ruester/git-po-de/wiki/Translation-Guidelines

ralfth commented on po/de.po:
```diff
@@ -12721,115 +12729,112 @@ msgstr "-s und -t mit beschädigten Objekten erlauben"
 #: builtin/cat-file.c:853
 msgid "like --batch, but don't emit <contents>"
-msgstr ""
+msgstr "wie --batch, aber keine Ausgabe von <Inhalte>"
```
`<Inhalten>`

ralfth commented on po/de.po:
```diff
@@ -12721,115 +12729,112 @@ msgstr "-s und -t mit beschädigten Objekten erlauben"
-#, fuzzy
 msgid "do not order objects before emitting them"
-msgstr "Konnte Objekt '%s' für symbolische Verknüpfung '%s' nicht lesen."
+msgstr "sortiere Objekte nicht, bevor sie ausgegeben werden"
```
`Objekte nicht ordnen, ...`

ralfth commented on po/de.po:
```diff
@@ -12940,7 +12945,7 @@ msgstr "alle Dateien im Index auschecken"
 #: builtin/checkout-index.c:221
 msgid "do not skip files with skip-worktree set"
-msgstr ""
+msgstr "keine Dateien mit gesetztem skip-worktree überspringen"
```
How about `Dateien mit gesetztem skip-worktree nicht überspringen`?

ralfth commented on po/de.po:
```diff
@@ -13069,7 +13074,7 @@ msgstr "'%s' oder '%s' kann nicht mit %s verwendet werden"
 #, c-format
 msgid "path '%s' is unmerged"
-msgstr "Pfad '%s' ist nicht zusammengeführt."
+msgstr "Pfad '%s' ist nicht zusammengeführtgit-sparse-checkou"
```
Something strange happened here.

ruester replied:
yeah, sorry for that :see_no_evil:

ralfth commented on po/de.po:
```diff
@@ -16525,125 +16530,116 @@ msgid "invalid parameter: expected sha1, got '%s'"
 #, c-format
 msgid "fsmonitor-daemon is watching '%s'\n"
-msgstr ""
+msgstr "fsmonitor-daemon beobachtet '%s '\n"
```
An additional space sneaked in.

ralfth commented on po/de.po:
```diff
@@ -16525,125 +16530,116 @@ msgid "invalid parameter: expected sha1, got '%s'"
 #: builtin/fsmonitor--daemon.c:1477
 msgid "fsmonitor--daemon not supported on this platform"
-msgstr ""
+msgstr "fsmonitor--daemon ist auf dieser Plattform nicht unterstützt"
```
`fsmonitor--daemon wird auf ...`

ralfth reviewed (changes requested):
Hi Matthias. Thank you for the translation! Please have a look at my comments for some messages.

PhillipSz commented on po/de.po:
```diff
@@ -2096,21 +2105,23 @@ msgid "not a valid branch point: '%s'"
 "You may try updating the submodules using 'git checkout %s && git submodule "
 "update --init'"
 msgstr ""
+"Sie können versuchen die Untermodule mit 'git checkout %s && git submodule "
```
Würde hier auch "Submodule" nutzen, wie sonst auch und das nicht übersetzten.

PhillipSz commented on po/de.po:
```diff
@@ -5670,9 +5683,8 @@ msgid "no pack files to index."
-#, fuzzy
 msgid "refusing to write multi-pack .bitmap without any objects"
-msgstr "Multipack-Bitmap konnte nicht geschrieben werden"
+msgstr "Schreiben des Multi-Pack-Bitmap ohne Objekte abgelehnt"
```
hmm müsste das nicht "der" Bitmap heißen?

PhillipSz commented on po/de.po:
```diff
@@ -11833,9 +11843,8 @@ msgid "visualize the bisection"
-#, fuzzy
 msgid "use <cmd>... to automatically bisect"
-msgstr "verwende <Programm>... für die automatische binäre Suche."
+msgstr "verwende <Programm>... für die automatische binäre Suche"
```
würde da glaube ich ein Leerschritt vor dem Punkt Punkt Punkt einbauen

ruester replied:
Das würde ich ungern vom ursprünglichen String abändern wollen. Hab mal geschaut: auch die anderen Übersetzer in anderen Sprachen ändern daran nichts, weil das vielleicht "mehrere Programme" bedeutet (?)...

PhillipSz commented on po/de.po:
```diff
@@ -18590,9 +18580,8 @@ msgid "use headers in message's body"
-#, fuzzy
 msgid "reading patches from stdin/tty..."
-msgstr "Aktualisierungen von der Standard-Eingabe lesen"
+msgstr "lese Patches von Standard-Eingabe/Terminal..."
```
würde hier ebenfalls Leerschritt machen vor Punkt Punkt Punkt

ruester replied:
Auch in anderen Übersetzungen haben wir es bisher so gehandhabt, z.b.:
```
msgid "First, rewinding head to replay your work on top of it...\n"
msgstr ""
"Zunächst wird der Branch zurückgespult, um Ihre Änderungen darauf neu "
"anzuwenden...\n"
```

Wenn ich in den anderen .po-Dateien reinschaue, dann wurde das auch dort nicht vom restlichen Text abgetrennt. Manche ändern es sogar zum UTF-8 Zeichen "…" (nur ein Zeichen im Vergleich zu drei: "...").
Es bedeutet im meisten Fall, dass auf etwas gewartet wird oder eine Berechnung länger dauer kann, also in dem Sinne keine Auslassung von etwas ist oder am Ende einer Aufzählung steht. Also falls wir das insgesamt ändern wollen, dann wäre das eine größere Änderung...

ruester replied:
CC @ralfth

PhillipSz replied:
Okay, gut können wir auch gerne so lassen wie es ist :)

PhillipSz commented on po/de.po:
```diff
@@ -21419,89 +21406,86 @@ msgid "you must specify a directory"
-#, fuzzy
 msgid "process the reflogs of all references"
-msgstr "das Überschreiben einer lokalen Referenz erzwingen"
+msgstr "Reflogs von allen referenzen verarbeiten"
```
Referenzen groß

PhillipSz reviewed (changes requested):
Hallo @ruester,
vielen Dank für die vielen Übersetzungen, habe nur ein paar Kommentare noch hinzugefügt.

Viele Grüße
Phillip

jiangxin committed:
Merge branch 'fz/po-zh_CN' of github.com:fangyi-zhou/git-po

* 'fz/po-zh_CN' of github.com:fangyi-zhou/git-po:
  l10n: zh_CN v2.36.0 round 1
  l10n: Update zh_CN repo link

jiangxin committed:
Merge branch 'fr_2.36_rnd1' of github.com:jnavila/git

* 'fr_2.36_rnd1' of github.com:jnavila/git:
  l10n: fr: v2.36 round 1

jnavila committed:
i18n: fix some badly formatted i18n strings

String in submodule--helper is not correctly formatting
placeholders. The string in git-send-email is partial.

Signed-off-by: Jean-Noël Avila <jn.avila@free.fr>
Signed-off-by: Junio C Hamano <gitster@pobox.com>

gitster committed:
Merge tag 'v2.35.2'

gitster committed:
Merge branch 'ja/i18n-fix-for-2.36'

Fixes to some localizable strings.

* ja/i18n-fix-for-2.36:
  i18n: fix some badly formatted i18n strings

gitster committed:
Git 2.36-rc2

Signed-off-by: Junio C Hamano <gitster@pobox.com>

ruester committed:
Apply changes from review and unify "multi-pack" and "split index"

Signed-off-by: Matthias Rüster <matthias.ruester@gmail.com>

ruester commented:
Many thanks for your review and comments. I now applied all your changes and also unified the translations for "multi-pack" and "split index".
@ralfth @PhillipSz Please have a look at this new commit and report back to me if something could be improved :wink: :+1:

PhillipSz reviewed (approved):
Sieht gut aus, danke!

ralfth reviewed (approved):
Looks good to me. Thank you!

Reviewed-by: Ralf Thielow <ralf.thielow@gmail.com>

jiangxin committed:
Merge branch 'master' of github.com:git/git

* 'master' of github.com:git/git: (25 commits)
  Git 2.36-rc2
  i18n: fix some badly formatted i18n strings
  Git 2.36-rc1
  t9902: split test to run on appropriate systems
  ls-tree doc: document interaction with submodules
  Documentation: add --batch-command to cat-file synopsis
  git-ls-tree.txt: fix the name of "%(objectsize:padded)"
  submodule-helper: fix usage string
  doc: replace "--" with {litdd} in credential-cache/fsmonitor
  contrib/scalar: fix 'all' target in Makefile
  Documentation/Makefile: fix "make info" regression in dad9cd7d518
  configure.ac: fix HAVE_SYNC_FILE_RANGE definition
  git-compat-util: really support openssl as a source of entropy
  ls-tree: `-l` should not imply recursive listing
  Git 2.35.2
  Git 2.34.2
  Git 2.33.2
  Git 2.32.1
  Git 2.31.2
  Git 2.30.3
  ...

jiangxin committed:
l10n: git.pot: v2.36.0 round 2 (4 new, 3 removed)

Generate po/git.pot from v2.36.0-rc2 for git v2.36.0 l10n round 2.

Signed-off-by: Jiang Xin <worldhello.net@gmail.com>

bitigchi committed:
l10n: tr: v2.36.0 round 2

Signed-off-by: Emir SARI <emir_sari@icloud.com>

ruester committed:
Update de.po for new translations

Signed-off-by: Matthias Rüster <matthias.ruester@gmail.com>

ruester committed:
l10n: de.po: Update German translation (round 2)

Signed-off-by: Matthias Rüster <matthias.ruester@gmail.com>

ruester commented:
Many thanks for your feedback 😃👍 

Today there was another translation update (round 2) with some new messages. You can see the new translations here: 25194d0af5

If this is also ok with you I will send all the translations to the upstream repository as a squashed commit tomorrow.

ralfth commented:
Looks good to me. Thanks!

Reviewed-by: Ralf Thielow <ralf.thielow@gmail.com>

PhillipSz commented:
Perfect, thanks!

Reviewed-by: Phillip Szelat phillip.szelat@gmail.com

ruester commented:
https://github.com/git-l10n/git-po/pull/622
